lib/rules/base.js

```diff
@@ -524,12 +515,34 @@ module.exports = class Base {
       rule: this.ruleName,
       severity: this.severity,
     };
+
     if (this._filePath) {
       defaults.filePath = this._filePath;
     }
+
     if (this[MODULE_NAME]) {
       defaults.moduleId = this[MODULE_NAME];
     }
+
+    // perform the node prop expansion only if those properties don't exist in result
+    if (result.node) {
```

Comment:
Let's leverage the `Object.assign` just below to simplify this a bit.

```js
let nodeExpansion = node && {
  line: node.loc && node.loc.start.line,
  column: node.loc && node.loc.start.column,
  source: this.sourceForNode(node),
};

let reportedResult = Object.assign({}, defaults, nodeExpansion, result);
```

Reply:
Updated with a few tweaks.